arrow/arrow.py

```diff
@@ -50,55 +111,77 @@ class Arrow:
 
     """
 
-    resolution = datetime.resolution
-
-    _ATTRS = ["year", "month", "day", "hour", "minute", "second", "microsecond"]
-    _ATTRS_PLURAL = [f"{a}s" for a in _ATTRS]
-    _MONTHS_PER_QUARTER = 3
-    _SECS_PER_MINUTE = float(60)
-    _SECS_PER_HOUR = float(60 * 60)
-    _SECS_PER_DAY = float(60 * 60 * 24)
-    _SECS_PER_WEEK = float(60 * 60 * 24 * 7)
-    _SECS_PER_MONTH = float(60 * 60 * 24 * 30.5)
-    _SECS_PER_YEAR = float(60 * 60 * 24 * 365)
+    resolution: ClassVar[timedelta] = dt_datetime.resolution
+    min: ClassVar["Arrow"]
+    max: ClassVar["Arrow"]
+
+    _ATTRS: Final[List[str]] = [
+        "year",
+        "month",
+        "day",
+        "hour",
+        "minute",
+        "second",
+        "microsecond",
+    ]
+    _ATTRS_PLURAL: Final[List[str]] = [f"{a}s" for a in _ATTRS]
+    _MONTHS_PER_QUARTER: Final[int] = 3
+    _SECS_PER_MINUTE: Final[float] = float(60)
+    _SECS_PER_HOUR: Final[float] = float(60 * 60)
+    _SECS_PER_DAY: Final[float] = float(60 * 60 * 24)
+    _SECS_PER_WEEK: Final[float] = float(60 * 60 * 24 * 7)
+    _SECS_PER_MONTH: Final[float] = float(60 * 60 * 24 * 30.5)
+    _SECS_PER_YEAR: Final[float] = float(60 * 60 * 24 * 365)
+
+    _SECS_MAP: Final[Mapping[TimeFrames, float]] = {
+        "second": 1.0,
+        "minute": _SECS_PER_MINUTE,
+        "hour": _SECS_PER_HOUR,
+        "day": _SECS_PER_DAY,
+        "week": _SECS_PER_WEEK,
+        "month": _SECS_PER_MONTH,
+        "year": _SECS_PER_YEAR,
+    }
+
+    _datetime: dt_datetime
```

Comment:
This isn't Class variable. It's a [Instance Variable](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#classes)

Reply:
I am pretty sure that variables declared outside of a function and within the class body are static/class variables: https://stackoverflow.com/questions/68645/are-static-class-variables-possible-in-python.

Reply:
If there are initial value, then the it becomes Class variable. but because I only declare variable without initialization, it is just type declaration for Instance variable. You can check [PEP-526](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations)
```python
Python 3.8.6 (default, Sep 30 2020, 04:00:38) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> arrow.Arrow._datetime
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'Arrow' has no attribute '_datetime'
>>> arrow.Arrow._SECS_MAP
{'second': 1.0, 'minute': 60.0, 'hour': 3600.0, 'day': 86400.0, 'week': 604800.0, 'month': 2635200.0, 'year': 31536000.0}
```

Reply:
Ah yes, fantastic!